manifests/config/queue.pp

```diff
@@ -0,0 +1,33 @@
+# Define: qpid::config::queue
+#
+# This define ensures an queue exists
+#
+# === Parameters
+#
+# $queue::                      Name of the queue
+#
+# $hostname::                   Hostname of the qpid broker
+#
+# $port::                       Port that qpid is listening on
+#
+# $ssl_cert::                   SSL cert to use for qpid-config commands
+#
+# $ssl_key::                    SSL key to use for qpid-config commands
+#
+define qpid::config::queue(
+  $queue = $title,
+  $hostname = undef,
+  $port = undef,
+  $ssl_cert = undef,
+  $ssl_key = undef,
+)
+{
+  qpid::config_cmd { "ensure queue ${queue}":
+    command  => "add queue ${queue} --durable",
```

Comment:
It's what puppet-katello does. I also thought about parametrizing this and wondered what the purpose of adding non-durable queues using puppet is.